.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "third_party/libipt"]
```

Comment:
Adding submodules in general brings up a bunch of auxiliary support we probably want.  I wonder if we should separate this out as its own PR.  I assume we want the submodule to be initialized all the time?  If so, things we would want:

+ We have to tell GA CI to init submodules: `submodules: true` arg to checkout step
+ Add submodule update to `make/git/devsetup.sh`
+ Add submodule update to `make/git/git_pullall.sh`
+ Potentially add to how-to-build docs if not completely automated
+ Potentially add a pre-commit hook to prevent the common mistake of accidentally rolling back a recent submodule update when switching branches

Reply:
We may also need to lock in what commit from the submodule repo we want to avoid future breakage from change made in the submodule.

Reply:
> We may also need to lock in what commit from the submodule repo we want to avoid future breakage from change made in the submodule.

The checked-out submodule will not change unless we tell it to so I don't think we need anything here.  We'll check in an update to the submodule explicitly when there's a bugfix or feature we want.

Reply:
> Adding submodules in general brings up a bunch of auxiliary support we probably want. I wonder if we should separate this out as its own PR. I assume we want the submodule to be initialized all the time? If so, things we would want:
> 
> * We have to tell GA CI to init submodules: `submodules: true` arg to checkout step
> * Add submodule update to `make/git/devsetup.sh`
> * Add submodule update to `make/git/git_pullall.sh`
> * Potentially add to how-to-build docs if not completely automated
> * Potentially add a pre-commit hook to prevent the common mistake of accidentally rolling back a recent submodule update when switching branches

I agree. I think use cmake to init & update the submodule is not perfect solution. I can create a new PR for this.